types/index.d.ts

```diff
@@ -0,0 +1,153 @@
+import type { Client } from 'aedes';
+import type { AedesPacket } from 'aedes-packet';
+import type { Readable } from 'stream';
+
+export type { AedesPacket as Packet } from 'aedes-packet';
+
+export interface AedesPersistenceSubscription {
+  clientId?: string;
+  topic: string;
+  qos: number;
+}
+
+type LastHearthbeatTimestamp = Date;
+
+export interface Brokers {
+  [brokerId: string]: LastHearthbeatTimestamp;
+}
+
+export type CallbackError = Error | null | undefined;
+
+export type WillPacket = AedesPacket & { [key: string]: any };
+
+interface Incoming {
+  [clientId: string]: { [messageId: string]: AedesPacket };
+}
+
+export class AedesMemoryPersistence {
+  private _retained: AedesPacket[];
+  private _subscriptions: Map<string, Map<string, number>>;
```

Comment:
Just for reference, is it possible here to be more specific about the key values? Even a comment could be useful

Reply:
yes something like this ? 
`Map<AedesPersistenceSubscription['clientId'], Map<AedesPersistenceSubscription['topic'], AedesPersistenceSubscription['qos']>>;`

Reply:
Not a types expert but if it works would be better yes :)